app/src/org/commcare/activities/FormHierarchyActivity.java

```diff
@@ -165,13 +187,18 @@ private void refreshView() {
         String hierarchyPath;
         try {
             hierarchyPath = FormHierarchyBuilder.populateHierarchyList(this, formList);
-        } catch (XPathTypeMismatchException e) {
+        } catch (XPathException e) {
             XPathErrorLogger.INSTANCE.logErrorToCurrentApp(e);
 
-            final String errorMsg = "Encounted xpath error: " + e.getMessage();
-            Toast.makeText(this, errorMsg, Toast.LENGTH_LONG).show();
-            setResult(RESULT_XPATH_ERROR);
-            finish();
+            final String title = StringUtils.getStringRobust(this, org.commcare.dalvik.R.string.error_occured);
+            final String errorMsg = e.getMessage();
+
+            AlertDialogFragment.fromCommCareAlertDialog(
```

Comment:
You should now be able to modify this to just call `UserfacingErrorHandling(). logErrorAndShowDialog` here instead of showing the dialog manually.